xc7/utils/prjxray_form_channels.py

```diff
@@ -929,6 +986,160 @@ def create_constant_tracks(conn):
     return create_track(vcc_node, unique_pos), \
            create_track(gnd_node, unique_pos)
 
+vpr_pkey = 999999 # This is a hack which ensures that pkeys of tiles in VPR
+                  # grid will differ from those in physical grid
+
+def insert_vpr_tile(conn, vpr_tile_name, vpr_tile_loc, vpr_tile_type_pkey, phy_tile_pkey):
+    """
+    Inserts a tile into the VPR tile grid. Adds also location correspondence
+    (through pkeys) to its counterpart in physical tile grid
+
+    Args:
+        conn: Database connection
+        vpr_tile_name: Tile name for VPR
+        vpr_tile_loc: Tile location in VPR grid (tuple)
+        vpr_tile_type_pkey: Tile type pkey
+        phy_tile_pkey: Corresponding tile pkey in the physical grid
+
+    Returns:
+        None
+    """
+
+    c = conn.cursor()
+
+    global vpr_pkey
+
+    # Insert the tile into the tile grid
+    c.execute(
+        "INSERT INTO tile(pkey, name, tile_type_pkey, grid_x, grid_y)"
+        "VALUES (?, ?, ?, ?, ?)",
+        (vpr_pkey, vpr_tile_name, vpr_tile_type_pkey, vpr_tile_loc[0], vpr_tile_loc[1])
```

Comment:
This is super super wrong.

Reply:
Why are you even assign a pkey?  Primary keys are generated automatically?

Reply:
Same as above